From 94e6da969265ef5cff93caaeb4c3955c27b0d8f2 Mon Sep 17 00:00:00 2001
From: Nicolas Capens <capn@google.com>
Date: Thu, 13 Nov 2014 22:55:51 -0500
Subject: [PATCH] Reference/release both the image and parent texture.

BUG=18381089

Change-Id: I379deb9a80eef4cc75082a58bfd14e588d7a93a1
---

diff --git a/src/OpenGL/libGLES_CM/Image.cpp b/src/OpenGL/libGLES_CM/Image.cpp
index 981cfc9..261ab90 100644
--- a/src/OpenGL/libGLES_CM/Image.cpp
+++ b/src/OpenGL/libGLES_CM/Image.cpp
@@ -54,7 +54,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->addRef();
+			parentTexture->addRef();
 		}
 
 		sw::atomicIncrement(&referenceCount);
@@ -64,7 +64,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->release();
+			parentTexture->release();
 		}
 
 		if(referenceCount > 0)
diff --git a/src/OpenGL/libGLESv2/Image.cpp b/src/OpenGL/libGLESv2/Image.cpp
index 2c5dcda..038d18e 100644
--- a/src/OpenGL/libGLESv2/Image.cpp
+++ b/src/OpenGL/libGLESv2/Image.cpp
@@ -53,7 +53,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->addRef();
+			parentTexture->addRef();
 		}
 
 		sw::atomicIncrement(&referenceCount);
@@ -63,7 +63,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->release();
+			parentTexture->release();
 		}
 
 		if(referenceCount > 0)
diff --git a/src/Radiance/libRAD/Image.cpp b/src/Radiance/libRAD/Image.cpp
index 4f41dc0..9b62b4b 100644
--- a/src/Radiance/libRAD/Image.cpp
+++ b/src/Radiance/libRAD/Image.cpp
@@ -53,7 +53,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->addRef();
+			parentTexture->addRef();
 		}
 
 		sw::atomicIncrement(&referenceCount);
@@ -63,7 +63,7 @@
 	{
 		if(parentTexture)
 		{
-			return parentTexture->release();
+			parentTexture->release();
 		}
 
 		if(referenceCount > 0)
